src/Api/TradingView/index.js

```diff
@@ -86,54 +86,67 @@ export const dataFeed = {
     },
 
     getBars: async (symbolInfo, resolution, periodParams, onResult, onErrorCallback) => {
-      const { from, to, firstDataRequest, countBack } = periodParams;
+      let { from, to, countBack } = periodParams;
       console.log('[getBars]: Method call', symbolInfo, resolution, periodParams);
       try {
         const symbol = symbolInfo.name.split("/")[0];
         const period = supportedResolutionsToPeriod[resolution];
 
-        if (countBack === 0 || !firstDataRequest) {
-          onResult([], { noData: true });
-        }
-
-        const prices = await getChartPrices(42161, symbol, period, { from, to });
-
-        if (prices.length === 0) {
-          // "noData" should be set if there is no data in the requested period.
-          onResult([], { noData: true });
-          return;
+        let cumulativePrices = [];
+        let prices = [];
+        const timeDiff = CHART_PERIODS[period] * 1000;
+        let count = 0;
+        while (cumulativePrices.length < countBack) {
+          if (count > 0) {
+            to = from;
+            from = from - (count * timeDiff)
+          }
+          prices = await getChartPrices(42161, symbol, period, { from, to });
+          console.log(`[getBars]: found ${prices.length} more bars, target: ${countBack} bars, cumulative total: ${cumulativePrices.length + prices.length} bars`);
+          if (!prices || prices.length === 0) {
+            break;
+          } else {
+            cumulativePrices = cumulativePrices.concat(prices)
+            count += 1;
+          }
         }
 
-        // const prices = fillGaps(prices_, CHART_PERIODS[period])
-
-        const bars = prices.map((el) => {
+        const uniqueBars = {};
+        fillGaps(cumulativePrices, CHART_PERIODS[period]).forEach((el) => {
           let low = el.low;
           if (low === 0) {
             low = el.open * 0.9996;
           }
-          return {
+          uniqueBars[el.time] = {
             ...el,
             low,
             time: el.time * 1000, //TradingView requires bar time in ms
-          };
+          }
         })
+        const bars = Object.values(uniqueBars).sort((a, b) => a.time - b.time);
+        console.log(`[getBars]: returned ${bars.length} bar(s), countBack: ${countBack}`);
+
+        onResult(bars, { noData: bars.length < countBack });
```

Comment:
is this what `noData` is supposed to be? `bars.length < countBack` looks like it would be something more like `isMoreDataToFetch`